crates/nu-cli/src/syntax_highlight.rs

```diff
@@ -37,97 +36,53 @@ impl Highlighter for NuHighlighter {
         );
 
         for shape in &shapes {
-            if shape.0.end <= last_seen_span
-                || last_seen_span < global_span_offset
-                || shape.0.start < global_span_offset
-            {
+            if shape.0.end <= last_seen_span || shape.0.start < global_span_offset {
```

Comment:
Just as a question for understanding what is the precondition that the middle OR condition can be elided?

Reply:
The initial thought was that `global_span_offset` is already guarded by `shape.0.start`.  So if the previous `shape` (and `last_seen_span = shape.0.end` during the iteration) was partially behind `global_span_offset`, it would've continued and wouldn't have changed `last_seen_span`.

However, thinking about it, I am unsure I can guarantee this is the case.  I think I will put the check back, in case there is some pathological case I am missing.

Honestly, this whole block confuses me.  `flatten_block` does this:

```Rust
    let mut output = vec![];

    for pipeline in &block.pipelines {
        output.extend(flatten_pipeline(working_set, pipeline));
    }
    output
```

So it _should_ be continuously going forward in span.  But I don't have enough expertise to prove or disprove that.

Reply:
I built Nushell without this `if ... continue` statement, and all of the highlighting I checked still works.  I believe I should restore this to its original form and not touch it, until I manage to figure out the intent behind it.